### PR TITLE
Disallow DDL of cipher column, except ADD

### PR DESCRIPTION
**Pull request by Pace2Car**
Changes proposed in this pull request:
  - Disallow some DDL of cipher column

---

Before committing this PR, I'm sure that I have checked the following options:
- [x] My code follows the [code of conduct](https://shardingsphere.apache.org/community/en/involved/conduct/code/) of this project.
- [x] I have self-reviewed the commit code.
- [ ] I have (or in comment I request) added corresponding labels for the pull request.
- [x] I have passed maven check locally : `./mvnw clean install -B -T1C -Dmaven.javadoc.skip -Dmaven.jacoco.skip -e`.
- [ ] I have made corresponding changes to the documentation.
- [x] I have added corresponding unit tests for my changes.

### CODE REVIEW

**Comment by Pace2Car** [comment]
image

File: features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/EncryptAlterTableTokenGenerator.java [hunk 1]
```diff
@@ -134,26 +134,13 @@ private Collection<SQLToken> getModifyColumnTokens(final EncryptTable encryptTab
         for (ModifyColumnDefinitionSegment each : segments) {
             String columnName = each.getColumnDefinition().getColumnName().getIdentifier().getValue();
             if (encryptTable.isEncryptColumn(columnName)) {
-                result.addAll(getModifyColumnTokens(encryptTable.getEncryptColumn(columnName), each));
+                throw new UnsupportedOperationException("Unsupported operation 'modify' for the cipher column");
```
Comment (RaigorJiang) [inline features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/EncryptAlterTableTokenGenerator.java:137]: Is it better to use ShardingSpherePreconditions?

File: features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/EncryptAlterTableTokenGenerator.java [hunk 2]
```diff
@@ -165,6 +152,10 @@ private Optional<SQLToken> getColumnPositionToken(final EncryptTable encryptTabl
     private Collection<SQLToken> getChangeColumnTokens(final EncryptTable encryptTable, final Collection<ChangeColumnDefinitionSegment> segments) {
         Collection<SQLToken> result = new LinkedList<>();
         for (ChangeColumnDefinitionSegment each : segments) {
+            String columnName = each.getPreviousColumn().getIdentifier().getValue();
+            if (encryptTable.isEncryptColumn(columnName)) {
+                throw new UnsupportedOperationException("Unsupported operation 'change' for the cipher column");
```
Comment (RaigorJiang) [inline features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/EncryptAlterTableTokenGenerator.java:157]: Is it better to use ShardingSpherePreconditions?

**Review by RaigorJiang — CHANGES_REQUESTED** [review]

File: features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/EncryptAlterTableTokenGenerator.java (hunk 2 above)
Comment (Pace2Car) [reply features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/EncryptAlterTableTokenGenerator.java]: THX for point out.

File: features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/EncryptAlterTableTokenGenerator.java (hunk 1 above)
Comment (Pace2Car) [reply features/encrypt/core/src/main/java/org/apache/shardingsphere/encrypt/rewrite/token/generator/EncryptAlterTableTokenGenerator.java]: THX for point out.

**Review by RaigorJiang — APPROVED** [review]

**Pull request merged**